tools/c7n_azure/c7n_azure/session.py

```diff
@@ -94,11 +109,13 @@ def _initialize_session(self):
         if self.credentials is None:
             self.log.error('Unable to locate credentials for Azure session.')
 
-    def client(self, client):
-        self._initialize_session()
+    def client(self, client, resource=None):
+        self._initialize_session(resource)
         service_name, client_name = client.rsplit('.', 1)
         svc_module = importlib.import_module(service_name)
         klass = getattr(svc_module, client_name)
+        if resource == 'https://graph.windows.net':
+            return klass(self.credentials, self.tenant_id)
```

Comment:
oh seriously? Yeah I guess we need the concept of subscription vs. tenant sessions.   We could put a bool, defaulted to false, on the client method, called "tenant_session" or something, might be cleaner than the string compare.  Will apply to a few different API's I guess.